contracts/btc-finality/src/finality.rs

```diff
@@ -29,6 +29,11 @@ use std::collections::{HashMap, HashSet};
 // or performance degradation caused by excessive future commitments.
 const MAX_PUB_RAND_COMMIT_OFFSET: u64 = 160_000;
 
+// Setting max amount of finalized blocks per EndBlock to 10_000 to cap processing time,
+// mirroring Babylon's `MaxFinalizedRewardedBlocksPerEndBlock`.
+// https://github.com/babylonlabs-io/babylon/blob/53d1a8e211f5c9d8b369397bde1f6cf05c7038ad/x/finality/types/constants.go#L7
+pub const MAX_FINALIZED_REWARDED_BLOCKS_PER_END_BLOCK: u64 = 10_000;
```

Comment:
We probably want a smaller value since CW operation is more expensive. Maybe 1000?

Reply:
Sounds good